Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp

```diff
@@ -117,8 +118,11 @@ DOM::ExceptionOr<JS::Value> XMLHttpRequest::response()
     }
     // 6. Otherwise, if this’s response type is "blob", set this’s response object to a new Blob object representing this’s received bytes with type set to the result of get a final MIME type for this.
     else if (m_response_type == Bindings::XMLHttpRequestResponseType::Blob) {
-        // FIXME: Implement this once we have 'Blob'.
-        return DOM::SimpleException { DOM::SimpleExceptionType::TypeError, "XHR Blob type not implemented" };
+        Vector<FileAPI::BlobPart> blob_parts;
+        blob_parts.append(adopt_ref(*new FileAPI::Blob { m_received_bytes, get_final_mime_type().type() }));
+        auto blob = FileAPI::Blob::create_with_global_object(verify_cast<Bindings::WindowObject>(global_object), blob_parts);
+        blob->set_wrapper(*Bindings::BlobWrapper::create(global_object, blob));
+        m_response_object = JS::make_handle(JS::Value(blob->wrapper()));
```

Comment:
Not sure if this incorrect or I'm doing something wrong. `blob-wrapper()` was returning a `nullptr` so I manually created a `BlobWrapper` and then called `blob->set_wrapper()`.